src/state/apis/swapper/helpers/validateTradeQuote.ts

```diff
@@ -299,7 +299,10 @@ export const validateTradeQuote = (
         },
       walletId &&
         !firstHopHasSufficientBalanceForGas && {
-          error: TradeQuoteValidationError.InsufficientFirstHopFeeAssetBalance,
+          error:
+            firstHopSellFeeAsset?.assetId === firstHop.sellAsset.assetId
+              ? TradeQuoteValidationError.InsufficientFirstHopAssetBalance
+              : TradeQuoteValidationError.InsufficientFirstHopFeeAssetBalance,
```

Comment:
~[blocking]: It seems this logic is flipped~ 
This logic is actually correct, we want to say ".... for gas" when trading ERC20s ✅
https://jam.dev/c/bf4d3496-106e-4d58-b09d-558ad4969fe1